### PR TITLE
Search

### PR DESCRIPTION
**Pull request by MollieS**

### CODE REVIEW

File: src/test/java/pkmncore/PokemonFinderTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import static org.junit.Assert.assertEquals;
+
+public class PokemonFinderTest {
+
+    private SearchEngine search = new SearchFake();
+    private PokemonFinder pokemonFinder = new PokemonFinder(search);
+
+    @Test
+    public void findsPokemonName() {
+        JsonObject data = search.findByName("pikachu");
```
Comment (gemcfadyen) [inline src/test/java/pkmncore/PokemonFinderTest.java]: I think you don't necessarily have to explicitly call the searchFake to get the expected object back before the assertion. 
You could do something like the following:

Pokemon pokemon = pokemonFinder.findName(data);
assertEquals("pikachu", pokemon.getName()); 

for example.

File: src/test/java/pkmncore/SearchTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+public class SearchTest {
+
+    @Test
+    public void getsAFakeJSONResponse() {
+        SearchFake fake = new SearchFake();
+        JsonObject response = fake.findByName("pika");
+        assertEquals("pikachu", response.get("name").getAsString());
+    }
+}
```
Comment (gemcfadyen) [inline src/test/java/pkmncore/SearchTest.java]: Here it looks like you are testing the Fake - we don't usually do this, unless there is a lot of `important` logic in the test double.

File: src/test/java/pkmncore/PokemonTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    @Test
+    public void hasAHeight() {
+        assertEquals(1, pokemon.getHeight());
+    }
+
+    @Test
+    public void hasAbilities() {
+        assertEquals("lightning-rod", pokemon.getAbilities()[0]);
```
Comment (gemcfadyen) [inline src/test/java/pkmncore/PokemonTest.java:24]: Could be worth checking the number of abilities returned is what is expected also.

**Comment by MollieS** [comment]
@gemcfadyen - thanks for comments, I have made suggested changes and will merge

**Pull request merged**